src/components/CreateAddress/PublicKeyImporter.jsx

```diff
@@ -51,6 +51,19 @@ class PublicKeyImporter extends React.Component {
     };
   }
 
+  componentDidUpdate(prevProps) {
+    const { number, setFinalized, addressType, publicKeyImporter } = this.props;
+    if (prevProps.addressType !== addressType) {
+      const publicKeyError = validatePublicKey(
+        publicKeyImporter.publicKey,
+        addressType
+      );
+      if (publicKeyError) {
```

Comment:
Can we avoid the nested if here? Maybe just do an `&& !validatePublicKey(...)` since it doesn't look like the actual error value is being used.